### DISCUSSION TITLE
GMSH parser error in line 533803642 in column 1 when using pumgen to convert msh2 file to puml.h5 file

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#925] [Q&A] [answered]
when I use `mpirun -np 48 pumgen loh.msh -s msh2` to convert msh2 file to puml.h5 file, I meet this problem, my gmsh version is 4.11.1, I'm not sure if it's the gmsh version that's causing this.
![image](https://github.com/SeisSol/SeisSol/assets/114474894/5515d09f-12e3-47ef-aea8-ed00d7010e14)

**Comment by sebwolf-de** [comment]
Can you post the first two lines of your gmsh file? They should look like
```
$MeshFormat
2.2 0 8
```

**Comment by buzhiqiuyuebai** [comment]
> Can you post the first two lines of your gmsh file? They should look like
> 
> ```
> $MeshFormat
> 2.2 0 8
> ```

I have added `Mesh.MshFileVersion = 2.2; `to the last line of the document.
![image](https://github.com/SeisSol/SeisSol/assets/114474894/a4783632-cbd6-4394-9bd6-2049a79a1b92)

**Comment by sebwolf-de** [comment]
Please show msh file around the line 533803642 then.

**Comment by buzhiqiuyuebai** [comment]
> Please show msh file around the line 533803642 then.

I think it caused by this line(I use this script to genetate meshes,and gmsh version is 4.11.1)
```
#!/bin/bash                  
#PBS -N abinit-test         
#PBS -l nodes=node02:ppn=2+node03:ppn=2+node04:ppn=2+node05:ppn=2+node06:ppn=4
#PBS -j oe                  
#PBS -q zhuosuan        
cd $PBS_O_WORKDIR
mpirun --host node02:2,node03:2,node04:2,node05:2,node06:4 gmsh -3 -nt 300 -algo hxt loh.geo
```
![image](https://github.com/SeisSol/SeisSol/assets/114474894/4b097372-c674-4f78-ba46-5c054d7e2151)

**Reply by sebwolf-de** [reply, marked as answer]
This line is clearly broken.
`27...` is the element number. 
`4` is the element type (tetrahedron)
`2` is the number of the additional tags
`1` and `2` are the additional tags
` 97...`, `21...`, `21...`, `21...` are the indices of the four nodes of the tetrahedron. 
Then there is another `4`, which together with the following data looks like another element. 
But a new element should be in its own line. Also this element misses its element id. 
In conclusion: The input file is not a correct msh2 file. Therefore, pumgen correctly throws an error message.

**Reply by krenzland** [reply]
By the way, GMSH doesn't really support MPI. At least the meshing won't go faster.

But yes, this is not a pumgen issue.
